QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/services/QueryService.java

```diff
@@ -1,30 +1,1220 @@
+import eu.neclab.ngsildbroker.commons.datatypes.terms.GeoQueryTerm;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.LanguageQueryTerm;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.QQueryTerm;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.ScopeQueryTerm;
+import eu.neclab.ngsildbroker.commons.datatypes.terms.TypeQueryTerm;
+import eu.neclab.ngsildbroker.commons.enums.ErrorType;
+import eu.neclab.ngsildbroker.commons.exceptions.ResponseException;
+import eu.neclab.ngsildbroker.commons.tools.EntityTools;
+import eu.neclab.ngsildbroker.commons.tools.HttpUtils;
+import eu.neclab.ngsildbroker.commons.tools.SerializationTools;
 import eu.neclab.ngsildbroker.queryhandler.repository.QueryDAO;
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.tuples.Tuple2;
+import io.smallrye.mutiny.tuples.Tuple3;
+import io.vertx.mutiny.core.MultiMap;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
+import io.vertx.mutiny.sqlclient.Row;
+import io.vertx.mutiny.sqlclient.RowIterator;
+import io.vertx.mutiny.sqlclient.RowSet;
+import io.vertx.mutiny.sqlclient.SqlConnection;
+
+@ApplicationScoped
+public class QueryService {
 
-@Singleton
-public class QueryService extends BaseQueryService {
+	private static Logger logger = LoggerFactory.getLogger(QueryService.class);
 
 	@Inject
 	QueryDAO queryDAO;
 
 	@Inject
-	CSourceDAO cSourceDAO;
+	Vertx vertx;
+
+	private WebClient webClient;
+
+	protected JsonLdOptions opts = new JsonLdOptions(JsonLdOptions.JSON_LD_1_1);
+
+	private Table<String, String, RegistrationEntry> tenant2CId2RegEntries = HashBasedTable.create();
+
+	@PostConstruct
+	void setup() {
+		webClient = WebClient.create(vertx);
+		queryDAO.getAllRegistries().onItem().transform(t -> {
+			tenant2CId2RegEntries = t;
+			return null;
+		}).await().indefinitely();
+	}
+
+	// This is needed so that @postconstruct runs on the startup thread and not on a
+	// worker thread later on
+	void startup(@Observes StartupEvent event) {
+	}
+
+	public Uni<QueryResult> query(String tenant, String qToken, boolean tokenProvided, String[] id,
+			TypeQueryTerm typeQuery, String idPattern, AttrsQueryTerm attrsQuery, QQueryTerm qQuery, CSFQueryTerm csf,
+			GeoQueryTerm geoQuery, ScopeQueryTerm scopeQuery, LanguageQueryTerm langQuery, int limit, int offSet,
+			boolean count, boolean localOnly, Context context) {
+		if (localOnly) {
+			return localQueryLevel1(tenant, id, typeQuery, idPattern, attrsQuery, qQuery, geoQuery, scopeQuery,
+					langQuery, limit, offSet, count);
+		}
+		if (!tokenProvided) {
+			return getAndStoreEntityIdList(tenant, id, idPattern, qToken, typeQuery, attrsQuery, geoQuery, qQuery,
+					scopeQuery, langQuery, context).onItem().transformToUni(t -> {
+						SqlConnection conn = t.getItem1();
+						EntityMap entityMap = t.getItem2();
+						List<EntityMapEntry> resultEntityMap = entityMap.getSubMap(offSet, limit + offSet);
```

Comment:
Please verify in other cases as limit and count cases.

Reply:
done